geode-core/src/main/java/org/apache/geode/management/internal/ContextAwareSSLRMIClientSocketFactory.java

```diff
@@ -44,18 +50,26 @@
   @Override
   public Socket createSocket(String host, int port) throws IOException {
     SocketCreator socketCreator;
+
     try {
       socketCreator =
           SocketCreatorFactory.getSocketCreatorForComponent(SecurableCommunicationChannel.JMX);
+      return socketCreator.connectForClient(host, port, 0);
     } catch (Exception exception) {
-      /*
-       * In the context of a gfsh VM the javax.net.ssl properties are used to configure SSL
-       * appropriately - see the constructor for JMXOperationInvoker.
-       */
-      return defaultFactory.createSocket(host, port);
+      try {
```

Comment:
I recommend extracting the nested try-catch to a new method. This example is small enough that it might not matter much, but usually that will be easier to read/maintain.

Reply:
Agreed. In this particular case (which is short and readable) though, I'd prefer to leave the implementation as it is right now.